src/main/java/study/step3/domain/RealCar.java

```diff
@@ -1,12 +1,14 @@
 package study.step3.domain;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 import java.util.Random;
 
 public class RealCar implements Car {
     private final String name;
-    private boolean moved;
     private final Random random;
```

Comment:
Car의 구현체가 생성될때마다 Random 객체가 만들어지는것도 괜찮지만
모든 RealCar들이 공유할 수 있도록 한번만 생성하고 캐싱하여 관리하는것도 어떨까요 :) ?
개인적으로 랜덤 객체는 RealCar의 상태 값으로 정의하기엔 조금 애매한것 같아요 ㅎㅎ

```suggestion
    private static final Random RANDOM = new Random();
```

Reply:
그렇네요. 👍